src/main.rs

```diff
@@ -11,25 +7,36 @@ use std::io::Write;
 use std::process;
 
 use gb_io::reader::SeqReader;
-use regex::Regex;
-
-// Translation table 11 from https://www.ncbi.nlm.nih.gov/Taxonomy/Utils/wprintgc.cgi.
-// Indexing into this array is as follows:
-// T = 0, C = 1, A = 2, G = 3
-// index = (16 * first_base) + (4 * second_base) + third_base
-// So... TTT = 0, TTC = 1, TTA = 2, ... , GGC = 61, GGA = 62, GGG = 63
-const GENETIC_CODE: &str = "FFLLSSSSYY**CC*WLLLLPPPPHHQQRRRRIIIMTTTTNNKKSSRRVVVVAAAADDEEGGGG";
-// some codons can be used as Met in the start position
-const STARTS: &str = "---M------**--*----M------------MMMM---------------M------------";
-
-// ERR_BAD_NT is an error value for an invalid nucleotide
+use gb_io::{feature_kind, qualifier_key};
+use once_cell::sync::Lazy;
+
+#[derive(thiserror::Error, Debug)]
+enum Error {
+    #[error("Something went wrong in translation! {0:?}")]
+    BadNucleotide(Vec<usize>),
+
+    #[error("Invalid amino acid: {0}")]
+    InvalidAminoAcid(char),
+}
+
+/// Translation table 11 from https://www.ncbi.nlm.nih.gov/Taxonomy/Utils/wprintgc.cgi.
+/// Indexing into this array is as follows:
+/// T = 0, C = 1, A = 2, G = 3
+/// index = (16 * first_base) + (4 * second_base) + third_base
+/// So... TTT = 0, TTC = 1, TTA = 2, ... , GGC = 61, GGA = 62, GGG = 63
+const GENETIC_CODE: &[u8] = b"FFLLSSSSYY**CC*WLLLLPPPPHHQQRRRRIIIMTTTTNNKKSSRRVVVVAAAADDEEGGGG";
+
+/// some codons can be used as Met in the start position
+const STARTS: &[u8] = b"---M------**--*----M------------MMMM---------------M------------";
+
+/// ERR_BAD_NT is an error value for an invalid nucleotide
 const ERR_BAD_NT: usize = 99;
 
-// map a base for indexing the GENETIC_CODE string
-// x (char): base to look up
-// returns: usize
-fn lookup(x: char) -> usize {
-    match x {
+/// map a base for indexing the GENETIC_CODE string
+/// x (char): base to look up
+/// returns: usize
```

Comment:
The extra `/` here might be a weird change, but it unlocks [Rust docstrings](https://doc.rust-lang.org/stable/book/ch14-02-publishing-to-crates-io.html?highlight=docu#making-useful-documentation-comments). This can generate HTML docs with `cargo doc`, and it is how the documentation in https://docs.rs/ is generated. This is how it looks like rendered:

![image](https://user-images.githubusercontent.com/6642/99850252-f3b72980-2b74-11eb-9f43-51288589b231.png)

Reply:
That's really neat, @luizirber , thanks! Can I publish these docs?

Reply:
you can run `cargo doc --no-deps` locally, and it will be available in `target/doc`. If you publish your package ("crate") to crates.io, it gets rendered automatically at https://docs.rs

You can also set up CI to render it with `cargo doc` and push it to gh-pages, I don't have an example at hand but it's certainly doable.